pages/community/tsc.js

```diff
@@ -0,0 +1,273 @@
+        </a>
+      </div>
+    </li>
+  );
+}
+
+function TwitterSVG() {
+  return (
+    <svg
+      className="w-5 h-5"
+      aria-hidden="true"
+      fill="currentColor"
+      viewBox="0 0 20 20"
+    >
+      <path d="M6.29 18.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0020 3.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.073 4.073 0 01.8 7.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.233 0 010 16.407a11.616 11.616 0 006.29 1.84" />
+    </svg>
+  );
+}
+
+function GithubSVG() {
+  return (
+    <svg
+      className="w-5 h-5"
+      fill="currentColor"
+      aria-hidden="true"
+      viewBox="0 0 30 30"
+    >
+      <path d="M 4.6582031 4 C 3.6182031 6.431 4.0382031 8.4602656 4.4082031 9.5722656 C 2.9152031 11.172266 2.0019531 13.3765 2.0019531 16.3125 C 2.0019531 23.5795 6.5919531 26 15.001953 26 C 23.411953 26 28.001953 23.5795 28.001953 16.3125 C 28.002953 13.4765 27.036656 11.264766 25.472656 9.6347656 C 25.696656 9.2667656 26.003906 8.5485 26.003906 7.1875 C 26.003906 5.1405 25.316406 4 25.316406 4 C 24.247406 4 21.735203 4.4310469 19.658203 6.4980469 C 18.183203 6.1310469 16.610906 6 15.003906 6 C 13.429906 6 11.888406 6.0862969 10.441406 6.4042969 C 10.348406 6.3182969 7.7612031 4 4.6582031 4 z M 10.005859 13.5 C 11.308859 13.514 13.458906 13.999 15.003906 14 C 16.548906 13.999 18.697 13.514 20 13.5 C 22.522 13.5 24 15.502656 24 17.972656 C 24 22.024656 21.656859 23.999 15.005859 24 L 15.003906 24 L 15 24 C 8.349 23.999 6.0058594 22.024656 6.0058594 17.972656 C 6.0058594 15.502656 7.4838594 13.5 10.005859 13.5 z M 10.5 17 A 1.5 2 0 0 0 9 19 A 1.5 2 0 0 0 10.5 21 A 1.5 2 0 0 0 12 19 A 1.5 2 0 0 0 10.5 17 z M 19.5 17 A 1.5 2 0 0 0 18 19 A 1.5 2 0 0 0 19.5 21 A 1.5 2 0 0 0 21 19 A 1.5 2 0 0 0 19.5 17 z" />
+    </svg>
+  );
+}
+
+function LinkedInSVG() {
+  return (
+    <svg
+      className="w-5 h-5"
+      aria-hidden="true"
+      fill="currentColor"
+      viewBox="0 0 20 20"
+    >
+      <path
+        fillRule="evenodd"
+        d="M16.338 16.338H13.67V12.16c0-.995-.017-2.277-1.387-2.277-1.39 0-1.601 1.086-1.601 2.207v4.248H8.014v-8.59h2.559v1.174h.037c.356-.675 1.227-1.387 2.526-1.387 2.703 0 3.203 1.778 3.203 4.092v4.711zM5.005 6.575a1.548 1.548 0 11-.003-3.096 1.548 1.548 0 01.003 3.096zm-1.337 9.763H6.34v-8.59H3.667v8.59zM17.668 1H2.328C1.595 1 1 1.581 1 2.298v15.403C1 18.418 1.595 19 2.328 19h15.34c.734 0 1.332-.582 1.332-1.299V2.298C19 1.581 18.402 1 17.668 1z"
+        clipRule="evenodd"
+      />
+    </svg>
+  );
+}
+
+function addAdditionalUserInfo(user) {
+  const userData = {
+    ...user,
+  };
+
+  addNameIfNotPresent(userData);
+  addSocialLinks(userData);
+  addAvatarUrl(userData);
+  userData.repos = userData.repos.map((repoName) =>
+    generateRepositoryInfo(repoName)
+  );
+
+  return userData;
+}
```

Comment:
Sorry 😅  It would be better to have this logic in one function and in several separate functions - at the end you only handle `user` object, so please make something like:

```js
function addAdditionalUserInfo(user) {
  const userData = {
    ...user,
  };
  
  if (!user.name) user.name = user.github;
  ... etc
}
```